### DISCUSSION TITLE
Get PHP 8 branch merged, and maybe drop PHP 5.6, 7.0, 7.1 and 7.2 support?

### DISCUSSION THREAD

**Discussion by elrido** [PrivateBin/PrivateBin#1117] [Polls]
TL;DR: If you use PrivateBin on PHP 5.6, 7.0, 7.1 or 7.2, could you provide an estimate year / month until when you consider you must keep using these versions? (i.e. provide the end-of-life date for the Linux distro you or your hoster are using)

After having recently upgraded my last PHP 7 development environment to a newer OS release, it has become difficult for me to do update the SRI hashes and other unit testing related tasks with the main branch. I would therefore like to merge the php8 branch, which contains updated unit tests that work with PHP 8 - this would allow me to do these tasks again more easily on all of my development devices.

As is explained over in [#707](https://github.com/PrivateBin/PrivateBin/issues/707) in more detail, the tests and PHP code is the same, but the PHP 8 compatible phpunit version 9 requires PHP 7.3 or higher. When we merge the php8 branch, we would no longer run unit tests on the older PHP versions. Of course at that point we knew that PrivateBin still works just fine even on the ancient PHP 5.6. But it is likely that as soon as we drop the tests on the old PHP releases, changes could start to creep in that inadvertently break support on these. So this might also be the point in time where the PrivateBin project officially drops support for these old PHP releases - but we don´t absolutely have to do so.

Hence this short questionnaire - depending of the outcome, I'll raise an according pull request.
